Branch-SDK/src/io/branch/referral/ServerRequest.java

```diff
@@ -262,25 +265,43 @@ private static ServerRequest getExtendedServerRequest(String requestPath, JSONOb
         return extendedReq;
     }
 
+    boolean skipOnTimeOut = false;
     /**
      * Updates the google ads parameters. This should be called only from a background thread since it involves GADS method invocation using reflection
      *
      * @param sysObserver {@link SystemObserver} instance.
      */
-    public void updateGAdsParams(SystemObserver sysObserver) {
-        try {
-            String advertisingId = sysObserver.getAdvertisingId();
-            if (advertisingId != null && getPost() != null) {
-                getPost().put(Defines.Jsonkey.GoogleAdvertisingID.getKey(), advertisingId);
-            }
-            int latVal = sysObserver.getLATValue();
-            if (getPost() != null) {
-                getPost().put(Defines.Jsonkey.LATVal.getKey(), latVal);
+    public void updateGAdsParams(final SystemObserver sysObserver) {
+        final CountDownLatch  latch  = new CountDownLatch(1);
+        skipOnTimeOut = false;
```

Comment:
This is to counter the case of calling "updateGAdsParams()" multiple time on same server request (Even though we  don't do it now)